src/03-ticket-calculator.js

```diff
@@ -54,12 +56,68 @@ const exampleTicketData = require("../data/tickets");
     calculateTicketPrice(tickets, ticketInfo);
     //> "Entrant type 'kid' cannot be found."
  */
-function calculateTicketPrice(ticketData, ticketInfo) {}
+function calculateTicketPrice(ticketData, ticketInfo) { // TODO - Refactor function into a switch statement if possible
+  let cost = 0
+
+  if (ticketInfo.ticketType === 'general') { //> General Admission
+    if (ticketInfo.entrantType === 'child') {
+      cost = ticketData.general.priceInCents.child
+    } else if (ticketInfo.entrantType === 'adult') {
+      cost = ticketData.general.priceInCents.adult
+    } else if (ticketInfo.entrantType === 'senior') {
+      cost = ticketData.general.priceInCents.senior
+    } else {
+      return `Entrant type '${ticketInfo.entrantType}' cannot be found.`
+    }
+   } else if (ticketInfo.ticketType === 'membership') { //> Membership Admission
+    if (ticketInfo.entrantType === 'child') {
+      cost = ticketData.membership.priceInCents.child
+    } else if (ticketInfo.entrantType === 'adult') {
+      cost = ticketData.membership.priceInCents.adult
+    } else if (ticketInfo.entrantType === 'senior') {
+      cost = ticketData.membership.priceInCents.senior
+    } else {
+      return `Entrant type '${ticketInfo.entrantType}' cannot be found.`
+    }
+  } else { //> Edge Case -- Invalid Ticket Type Information
+    return `Ticket type '${ticketInfo.ticketType}' cannot be found.`
+  }
+
+  // Extras
+  if (ticketInfo.extras.length > 0) { //> If an incorrect extra is provided, returns an error
+    if (ticketInfo.extras.includes('movie') === false && ticketInfo.extras.includes('education') === false && ticketInfo.extras.includes('terrace') === false) {
+      return `Extra type '${ticketInfo.extras[0]}' cannot be found.`
+    }
+  }
+  
+  if (ticketInfo.extras.includes('movie')) {
+    cost += 1000
+  }
+
+  if (ticketInfo.extras.includes('education')) {
+    if (ticketInfo.entrantType === 'child') {
+      cost += 1000
+    } else {
+      cost += 1200
+    }
+  }
+
+  if (ticketInfo.extras.includes('terrace')) {
+    if (ticketInfo.entrantType === 'child') {
+      cost += 500
+    } else {
+      cost += 1000
+    }
+  }
+
+  return cost
+} 
```

Comment:
This code works but there it can definitely be optimized and made more efficient. In instances like this you want to leverage your variables to check if entrantType, ticketType, or extras exist within the the `ticketData` object. For instance if you are checking the `ticketType ` that might look like 
```
if( !ticketData[ticketInfo.ticketType] ){
   return `Ticket type '${ticketInfo.ticketType}' cannot be found.`;
}
```
And getting the price for the `entrantType` might look like `const price = ticketData[ticketInfo.ticketType].priceInCents[ticketInfo.entrantType];`